File: OLSource1/v1/VS_officedev/codesnippet/CSharp/how-to--cache-data-in-a-password-protected-document_2.cs
        [CachedAttribute]
        public string CachedString = "This string is cached in the workbook.";

        private bool protectStructureValue;
        private bool protectWindowsValue;

        protected override void UnprotectDocument()
        {
            protectStructureValue = this.ProtectStructure;
            protectWindowsValue = this.ProtectWindows;

            this.Unprotect(securelyStoredPassword);
        }

        protected override void ProtectDocument()
        {
            this.Protect(securelyStoredPassword, protectStructureValue,
                protectWindowsValue);
        }

File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/how-to--cache-data-in-a-password-protected-document_2.vb
    <CachedAttribute()> _
    Public CachedString As String = "This string is cached in the workbook."

    Private protectStructureValue As Boolean
    Private protectWindowsValue As Boolean

    Protected Overrides Sub UnprotectDocument()
        protectStructureValue = Me.ProtectStructure
        protectWindowsValue = Me.ProtectWindows

        Me.Unprotect(securelyStoredPassword)
    End Sub

    Protected Overrides Sub ProtectDocument()
        Me.Protect(securelyStoredPassword, protectStructureValue, _
            protectWindowsValue)
    End Sub